### PR TITLE
Update triton-go library and upgrade to the new API

### PR DESCRIPTION
**Pull request by cheapRoc**
Our new `triton-go` API was released earlier in the week. We've centered around a new API and I've updated the Triton provider to accommodate this change set.

This PR includes the following...
* Updated vendor library of `triton-go`
* Refactored resources around the new API
* Introduce `make help` and `make vendor-triton`
* Cleaned up the `README.md`

This PR does not include any bug fixes or other changes, just support for the latest API. We'll address those concerns shortly.

### CODE REVIEW

File: triton/resource_machine.go [hunk 1]
```diff
@@ -450,7 +466,7 @@ func resourceMachineUpdate(d *schema.ResourceData, meta interface{}) error {
 	if d.HasChange("package") {
 		newPackage := d.Get("package").(string)
 
-		err := client.Machines().ResizeMachine(context.Background(), &triton.ResizeMachineInput{
+		err := c.Instances().Resize(context.Background(), &compute.ResizeInstanceInput{
```
Comment (sean-) [inline triton/resource_machine.go]: @jen20 'ya know, it is going to bug me that inputs are all allocated on the heap vs put on the stack.

File: triton/resource_fabric.go [hunk 2]
```diff
@@ -110,7 +114,7 @@ func resourceFabricCreate(d *schema.ResourceData, meta interface{}) error {
 		routes[cidr] = ip
 	}
 
-	fabric, err := client.Fabrics().CreateFabricNetwork(context.Background(), &triton.CreateFabricNetworkInput{
+	fabric, err := n.Fabrics().CreateFabric(context.Background(), &network.CreateFabricInput{
```
Comment (cheapRoc) [inline triton/resource_fabric.go]: This name originates as `CreateFabricNetwork`. I guess `CreateVXLAN` is by definition what we're actually performing on the backend. We already have [`CreateVLAN`](https://github.com/joyent/triton-go/blob/master/network/fabrics.go#L55) so I'm wondering if `n.Fabrics().CreateNetwork()` is more legible (or adds to confusion).

Comment (cheapRoc) [inline triton/resource_fabric.go]: Never mind, it looks like I was removing the stutter completely and missed a few functions. I've got `Fabrics().List()` but `Fabrics().GetFabric()`, `CreateFabric()`, `DeleteFabric()`. I can continue along those lines as originally intended.

Nice find!

**Comment by cheapRoc** [comment]
I get that, this legacy can be removed later and aligned with docs as well.

**Comment by cheapRoc** [comment]
Actually, I'm not seeing this in master. Is anyone else?

**Comment by cheapRoc** [comment]
Never mind... derp on my part.

**Pull request merged**